### ISSUE TITLE
🐛 Bug Report: association properties are not propagated in LangChain

### ISSUE THREAD

**Issue by eliasecchig** [traceloop/openllmetry#1921]
### Which component is this bug for?

Traceloop SDK

### 📜 Description

I noticed that [association properties](https://www.traceloop.com/docs/openllmetry/tracing/association) are not being added anymore to the span.
Last version I tested where attributes are working correctly is `0.25.6.`.

### 👟 Reproduction steps

1. Set some association properties
```
    Traceloop.set_association_properties({
        "log_type": "tracing",
        "commit_sha": "123"
    })
```
2. Execute a Langchain chain
3. Inspect the generated Span

### 👍 Expected behavior

Association properties should be under `traceloop.association.properties`

### 👎 Actual Behavior with Screenshots

```json
// Span 1 (Version 25.6 - Includes Association Properties)
{
  "name": "RunnableSequence.workflow",
  "context": {
    "trace_id": "0x1c338549663b190a7984a5be08a864d0",
    "span_id": "0x877d2367a723d954",
    "trace_state": "[]"
  },
  "kind": "SpanKind.INTERNAL",
  "parent_id": null,
  "start_time": "2024-08-30T15:31:14.374304Z",
  "end_time": "2024-08-30T15:31:16.619049Z",
  "status": {
    "status_code": "UNSET"
  },
  "attributes": {
    "traceloop.association.properties.log_type": "tracing", 
    "traceloop.association.properties.commit_sha": "123",
    "traceloop.span.kind": "workflow",
    "traceloop.entity.name": "RunnableSequence.workflow",
    "traceloop.entity.input": "{\"inputs\": {\"input\": \"\"}, \"tags\": [], \"metadata\": {}, \"kwargs\": {\"run_type\": null, \"name\": \"RunnableSequence\"}}",
    "traceloop.entity.output": "{\"outputs\": \"content=\\\"Hi there! \\ud83d\\udc4b  What kind of recipe are you looking for today? \\ud83d\\ude0a  Tell me what you're in the mood for, and I'll whip up some delicious ideas for you! \\ud83c\\udf73 \\ud83c\\udf72 \\ud83c\\udf70 \\\\n\\\" response_metadata={'is_blocked': False, id='run-8a01a905-b10e-443f-8fd4-95fc4e8a7752' usage_metadata={'input_tokens': 18, 'output_tokens': 48, 'total_tokens': 66}", \"kwargs\": {\"tags\": [], \"inputs\": {\"messages\": [{\"content\": [{\"type\": \"text\", \"text\": \"hi\"}], \"additional_kwargs\": {}, \"id\": null, \"name\": null, \"type\": \"human\", \"response_metadata\": {}, \"example\": false}], \"user_id\": \"my_user\", \"session_id\": \"197e33c1-f4cd-4bfb-ad29-05f92c5626aa\"}}}"
  },
  "events": [],
  "links": [],
  "resource": {
    "attributes": {
      "service.name": "Langchain Chatbot Application"
    },
    "schema_url": ""
  }
}


// Span 2 (Version 29.2 - Missing Association Properties)
{
  "name": "RunnableSequence.workflow",
  "context": {
    "trace_id": "0xae7ec55751f0613e5b4fe1a3b909543f",
    "span_id": "0x38584e669b9e3257",
    "trace_state": "[]"
  },
  "kind": "SpanKind.INTERNAL",
  "parent_id": null,
  "start_time": "2024-08-30T15:28:38.108989Z",
  "end_time": "2024-08-30T15:28:40.558043Z",
  "status": {
    "status_code": "UNSET"
  },
  "attributes": {
    "traceloop.workflow.name": "RunnableSequence",
    "traceloop.entity.path": "",
    "traceloop.span.kind": "workflow",
    "traceloop.entity.name": "RunnableSequence",
    "traceloop.entity.input": "{\"inputs\": {\"input\": \"\"}, \"tags\": [], \"metadata\": {}, \"kwargs\": {\"run_type\": null, \"name\": \"RunnableSequence\"}}",
    "traceloop.entity.output": "{\"outputs\": \"content=\\\"Hi there! \\ud83d\\udc4b  What kind of recipe are you looking for today? \\ud83d\\ude0a  Tell me what you're in the mood for, and I'll whip up some delicious ideas for you! \\ud83c\\udf73 \\\\n\\\" response_metadata={'is_blocked': False, id='run-75c269e5-7dc7-4adb-8b1e-76d475e134b4' usage_metadata={'input_tokens': 18, 'output_tokens': 44, 'total_tokens': 62}", \"kwargs\": {\"tags\": [], \"inputs\": {\"messages\": [{\"content\": [{\"type\": \"text\", \"text\": \"hi\"}], \"additional_kwargs\": {}, \"id\": null, \"name\": null, \"type\": \"human\", \"response_metadata\": {}, \"example\": false}], \"user_id\": \"my_user\", \"session_id\": \"e666c5e8-133b-4d10-b00e-4258c802901f\"}}}"
  },
  "events": [],
  "links": [],
  "resource": {
    "attributes": {
      "service.name": "Langchain Chatbot Application"
    },
    "schema_url": ""
  }
}
```

### 🤖 Python Version

3.10

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by nirga** [comment]
I suspect this is relate to #1541. We use python context to propagate these and I think LangChain callbacks are called from a different context

**Comment by nirga** [comment]
This seems to be introduced by #1805

**Comment by nirga** [comment]
Fixed with #1922
